inter_company_rules/models/purchase_order.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models, _
+from openerp.exceptions import Warning as UserError
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    auto_generated = fields.Boolean(string='Auto Generated Purchase Order',
+                                    copy=False)
+    auto_sale_order_id = fields.Many2one('sale.order',
+                                         string='Source Sale Order',
+                                         readonly=True, copy=False)
+
+    @api.multi
+    def button_confirm(self):
+        """ Generate inter company sale order base on conditions."""
+        res = super(PurchaseOrder, self).button_confirm()
+        for order in self:
+            # get the company from partner then trigger action of
+            # intercompany relation
+            company_rec = self.env['res.company']._find_company_from_partner(
+                order.partner_id.id)
+            if company_rec and company_rec.so_from_po and (
+                    not order.auto_generated):
+                order.inter_company_create_sale_order(company_rec)
+        return res
+
+    @api.one
+    def inter_company_create_sale_order(self, company):
+        """ Create a Sale Order from the current PO (self)
+            Note : In this method, reading the current PO is done as sudo,
+            and the creation of the derived
+            SO as intercompany_user, minimizing the access right required
+            for the trigger user.
+            :param company : the company of the created PO
+            :rtype company : res.company record
+        """
+        SaleOrder = self.env['sale.order']
+
+        # find user for creating and validation SO/PO from partner company
+        intercompany_uid = (company.intercompany_user_id and
```

Comment:
This can be simplify in this version: `intercompany_uid = company.intercompany_user_id.id`